### PR TITLE
Add custom network behaviour to propagation network

### PR DESCRIPTION
**Pull request by lemonholic**
- Renamed `simperby_network::propnet` to `simperby_network::propagation_network`
- Collected `Identify`, `Kademlia`, `Floodsub` into `pub struct Behaviour` to make a custom network behaviour.
  (A network behaviour defines what bytes to send and to whom)
- Added `pub struct Behaviour` to `propagation_network::PropagationNetwork`

### CODE REVIEW

File: network/src/propagation_network/mod.rs [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-/// The backbone network of simperby that propagates serialized data such as blocks and votes.
-/// This network discovers peers with Kademlia([`libp2p::kad`]),
-/// and propagates data with FloodSub([`libp2p::floodsub`]).
-pub struct PropagationNetwork {}
+mod behaviour;
+use behaviour::Behaviour;
+
+/**
+The backbone network of simperby that propagates serialized data such as blocks and votes.
+This network discovers peers with Kademlia([`libp2p::kad`]),
```
Comment (junha1) [inline network/src/propagation_network/mod.rs:14]: ditto.

File: network/src/propagation_network/mod.rs [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+and propagates data with FloodSub([`libp2p::floodsub`]).
+*/
+pub struct PropagationNetwork {
+    /// A custom network behaviour that collects other [`libp2p`] network behaviours
+    /// to extend their functionalities.
+    /// It implements [`libp2p::swarm::NetworkBehaviour`] as well as its field members.
+    _behaviour: Mutex<Behaviour>,
```
Comment (junha1) [inline network/src/propagation_network/mod.rs:21]: Ditto.

**Review by junha1 — COMMENTED** [review]
I think it's better to take a deeper look of your code after you present the actual use of `Behaviour`. Let's just merge this PR after you fix the minor doc issues I commented.

File: network/src/propagation_network/behaviour.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    floodsub::{Floodsub, FloodsubEvent},
+    identify::{Identify, IdentifyEvent},
+    kad::{store::MemoryStore, Kademlia, KademliaEvent},
+    NetworkBehaviour,
+};
+
+#[derive(NetworkBehaviour)]
+#[behaviour(out_event = "Event")]
+pub struct Behaviour {
+    /// A network behaviour that identifies connected peers.
```
[the remark this answers is not in the document]
Comment (lemonholic) [reply network/src/propagation_network/behaviour.rs:11]: fixed

File: network/src/propagation_network/mod.rs (hunk 1 above)
Comment (lemonholic) [reply network/src/propagation_network/mod.rs:12]: fixed

File: network/src/propagation_network/mod.rs (hunk 2 above)
Comment (lemonholic) [reply network/src/propagation_network/mod.rs:20]: fixed

**Review by junha1 — APPROVED** [review]

**Pull request merged**